internal/gribi/gribi.go

```diff
@@ -0,0 +1,203 @@
+	t.Helper()
+	t.Logf("setting GRIBI Election ID for dut: %s to low=%d,high=%d", g.dut.Name(), lowElecId, highElecId)
+	g.fluentC.Modify().UpdateElectionID(t, lowElecId, highElecId)
+	err := g.AwaitTimeout(context.Background(), t, timeout)
+	if err != nil {
+		t.Fatalf("learnElectionID Error: %v", err)
+	}
+	chk.HasResult(t, g.fluentC.Results(t),
+		fluent.OperationResult().
+			WithCurrentServerElectionID(lowElecId, highElecId).
+			AsResult(),
+	)
+	results := g.fluentC.Results(t)
+	mu.Lock()
+	defer mu.Unlock()
+	g.lastElectionID = results[len(results)-1].CurrentServerElectionID
+}
+
+// GetLastElectionID returnes the latest election id that is cached without querying it from dut.
+func (g *GRIBIHandler) GetLastElectionID(t testing.TB) (low, high uint64) {
+	return g.lastElectionID.Low, g.lastElectionID.High
+}
+
+// BecomeLeader learns the latest election id and the make the client leader by increasing the election id by one.
+func (g *GRIBIHandler) BecomeLeader(t testing.TB) {
+	t.Logf("trying to be a master with increasing the election id by one on dut: %s", g.dut.Name())
+	lowElecId, highElecId := g.LearnElectionID(t)
+	if lowElecId == maxUint64 {
+		highElecId = highElecId + 1
+	} else {
+		lowElecId = lowElecId + 1
+	}
+	g.UpdateElectionID(t, lowElecId, highElecId)
+}
+
+// AddNHG adds a NextHOP group entry
+func (g *GRIBIHandler) AddNHG(t testing.TB, nhgIndex uint64, nhWeights map[uint64]uint64, instance string, expectedResult fluent.ProgrammingResult) {
+	NGH := fluent.NextHopGroupEntry().WithNetworkInstance(instance).WithID(nhgIndex)
+	for nhIndex, weight := range nhWeights {
+		NGH.AddNextHop(nhIndex, weight)
+	}
+	g.fluentC.Modify().AddEntry(t, NGH)
+	if err := g.AwaitTimeout(context.Background(), t, timeout); err != nil {
+		t.Fatalf("got unexpected error from server adding NGH, got: %v, want: nil", err)
+	}
+
+	chk.HasResult(t, g.fluentC.Results(t),
+		fluent.OperationResult().
+			WithNextHopGroupOperation(nhgIndex).
+			WithOperationType(constants.Add).
+			WithProgrammingResult(expectedResult).
+			AsResult(),
+		chk.IgnoreOperationID(),
+	)
+}
+
+// AddNH adds a NextHOP entry
+func (g *GRIBIHandler) AddNH(t testing.TB, nhIndex uint64, address, instance string, expectedResult fluent.ProgrammingResult) {
+	g.fluentC.Modify().AddEntry(t,
+		fluent.NextHopEntry().
+			WithNetworkInstance(instance).
+			WithIndex(nhIndex).
+			WithIPAddress(address))
+
+	if err := g.AwaitTimeout(context.Background(), t, timeout); err != nil {
+		t.Fatalf("got unexpected error from server adding NH, got: %v, want: nil", err)
+	}
+	chk.HasResult(t, g.fluentC.Results(t),
+		fluent.OperationResult().
+			WithNextHopOperation(nhIndex).
+			WithOperationType(constants.Add).
+			WithProgrammingResult(expectedResult).
+			AsResult(),
+		chk.IgnoreOperationID(),
+	)
+}
+
+// AddNH adds an IPV4 entry
+func (g *GRIBIHandler) AddIPV4Entry(t testing.TB, nhgIndex uint64, nhgInstance, prefix, instance string, expectedResult fluent.ProgrammingResult) {
```

Comment:
Reorder the arguments so the mapping relationship is more clear?

```
func (g *GRIBIHandler) AddIPv4(t testing.TB, prefix string, nhgIndex uint64, instance, nhgInstance string, expectedResult fluent.ProgrammingResult)
```

Reply:
done